packages/atomic/src/components/common/smart-snippets/atomic-smart-snippet-collapse-wrapper/atomic-smart-snippet-collapse-wrapper.tsx

```diff
@@ -0,0 +1,82 @@
+import {Component, h, Element, State, Prop, Host} from '@stencil/core';
+import ArrowDown from '../../../../images/arrow-down.svg';
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {AnyBindings} from '../../interface/bindings';
+
+@Component({
+  tag: 'atomic-smart-snippet-collapse-wrapper',
+  styleUrl: 'atomic-smart-snippet-collapse-wrapper.pcss',
+  shadow: true,
+})
+export class AtomicSmartSnippetCollapseWrapper {
+  @InitializeBindings() public bindings!: AnyBindings;
+  public error!: Error;
+  @Element() public host!: HTMLElement;
+
+  @Prop({reflect: true}) public maximumHeight?: number;
+
+  @Prop({reflect: true}) public collapsedHeight?: number;
+
+  @State() public isExpanded = true;
+
+  @State() public showButton = true;
+
+  @State() private fullHeight?: number;
+
+  private shouldRenderButton = !!this.maximumHeight;
+
+  public componentDidRender() {
+    if (this.fullHeight === undefined && this.shouldRenderButton) {
```

Comment:
I do agree with your points, but I have a few questions: 

1. Since these props are directly set in the html and are not updated by anything in the code, wouldn't changing them cause a re-render (ergo no need to watch them specifically) ? I'm not too familiar with stencil sadly....
2. `host.getBoundingClientRect().height` is indeed used to check wether we need to collapse the component or not. However, using a resize observer to check the value returned by that function would cause an infinite loop, as the value changes when collapsing the smart snippet (initial render => calculate full height => collapse => observer triggers => recalculate full height => uncollapse => recalculate => etc). This is why, like in the `atomic-smart-snippet-expendable-answer` component, the `initializeFullHeight` function is only ran once on component initialization, and not after (we just need to calculate the theoretical full height in order to asses wether the collapse logic should be activated or not). 

I'll await your advice regarding these 2 points. If listeners on the props are needed i'll add them, however I don't think the observer is necessary !

Reply:
Ok I see. Then I guess there should be no check on `this.fullHeight === undefined` ?

Reply:
Hmmm, according to [this](https://stenciljs.com/docs/component-lifecycle#componentdidrender) didRender re-runs everytime the render function executes. In my case, on the first render, I'm checking whether the collapse feature needs to be activated, then update values accordingly, which will definitely cause a re-render. If I don't use `=== undefined` I'm once again at risk of an infinite loop which toggles the feature on and off since, as mentioned before, recalculating the high of the host will give a different value once the feature is activated. 

TLDR; in need this so that it gets run only on the first render to avoid an infinite loop !